compiler/noirc_frontend/src/hir/comptime/interpreter.rs

```diff
@@ -328,21 +328,27 @@ impl<'local, 'interner> Interpreter<'local, 'interner> {
                 let argument = Value::Pointer(Shared::new(argument), true);
                 self.define_pattern(pattern, typ, argument, location)
             }
-            HirPattern::Tuple(pattern_fields, _) => match (argument, typ) {
-                (Value::Tuple(fields), Type::Tuple(type_fields))
-                    if fields.len() == pattern_fields.len() =>
-                {
-                    for ((pattern, typ), argument) in
-                        pattern_fields.iter().zip(type_fields).zip(fields)
+            HirPattern::Tuple(pattern_fields, _) => {
+                let typ = &typ.follow_bindings();
```

Comment:
The `as_integer` method didn't work without this. I would get an error saying "Expected (u8, bool) but get (u8, bool)". Debugging it, `typ` was a type variable. Let me know if this was the right thing to do.

Reply:
This is good, we should _always_ call `follow_bindings` before matching on a specific variant of a type for the exact reason you said. It was a mistake this wasn't done originally.